Sources/PackageDescription/Target.swift

```diff
@@ -981,15 +986,25 @@ extension Target.Dependency {
     ///   - package: The name of the package.
     ///   - condition: A condition that limits the application of the target dependency. For example, only apply a
     ///       dependency for a specific platform.
-    @available(_PackageDescription, introduced: 5.3)
+    @available(_PackageDescription, introduced: 5.3, obsoleted: 999.0)
     public static func product(
         name: String,
         package: String,
         condition: TargetDependencyCondition? = nil
     ) -> Target.Dependency {
-        return .productItem(name: name, package: package, condition: condition)
+        return .productItem(name: name, package: package, moduleAliases: nil, condition: condition)
     }
-    
+
+    @available(_PackageDescription, introduced: 999.0)
+    public static func product(
```

Comment:
needs API doc

Reply:
Should we call this new parameter `targetAliases` instead of `moduleAliases` btw? Any opinions on this @neonichu @abertelrud @tomerd ?

Reply:
No, I think `moduleAliases` is the best, since not all targets produce modules and the alias here is really for the module name, not the target name.  We should actually have a parameter to optionally control the module name here (right now it's just always mangled from the target name) but that can be added separately.